lms/views/basic_lti_launch.py

```diff
@@ -65,15 +64,17 @@ def canvas_file_basic_lti_launch(self):
         Via. We have to re-do this file-ID-for-download-URL exchange on every
         single launch because Canvas's download URLs are temporary.
         """
-        canvas_api_client = self.request.find_service(name="canvas_api_client")
         file_id = self.request.params["file_id"]
 
-        try:
-            document_url = canvas_api_client.public_url(file_id)
-        except CanvasAPIError:
-            return {}
+        self.context.js_config["urls"].update(
+            {
+                "via_url": self.request.route_url(
+                    "canvas_api.files.via_url", file_id=file_id
```

Comment:
So rather than reading the `file_id` from `js-config` and constructing the URL to the `via_url` API, the frontend just reads the entire URL from `js-config`. We should fix it to do this for the `list_files` API and for the popup window's URL (complete with `authorization` query param) as well.

The backend _could_ even add the `authorization` query param to the `via_url` here, and then the frontend would not need to read `authToken` from the config and add it as a header.

Reply:
From the front-end's point of view, the most important thing is that API calls behave consistently. For that reason, either the auth should always be baked into the URL or never. I think my preference is to stick with the Authorization header for general consistency across projects making authenticated calls to some kind of API.